lib/cacheable/controller.rb

```diff
@@ -37,10 +37,16 @@ def response_cache(key_data=nil, version_data=nil, &block)
       end
 
       handler = Cacheable::ResponseCacheHandler.new(self) do |h|
-        h.key_data       = key_data       || cache_key_data
-        h.version_data   = version_data   || cache_version_data
-        h.block          = block
-        h.cache_store    = Cacheable.cache_store
+        h.key_data                                       = key_data       || cache_key_data
+        h.version_data                                   = version_data   || cache_version_data
+        h.block                                          = block
+        h.cache_store                                    = Cacheable.cache_store
+        h.env                                            = request.env
+        h.cache_age_tolerance                            = cache_age_tolerance_in_seconds
+        h.serve_unversioned                              = serve_unversioned_cacheable_entry?
+        h.force_refill_cache                             = force_refill_cache?
+        h.cache_hit_server                               = ->(status, _, body) { render status: status, plain: body }
```

Comment:
WDYT about returning the status from the `ResponseCacheHandler` and having the caller run the code required to serve each option?

Reply:
I think I like that better 👍